network/src/processor/js/graphql_api_v1.rs

```diff
@@ -0,0 +1,160 @@
+// Copyright 2020 ChainSafe Systems
+// SPDX-License-Identifier: Apache-2.0
+
+//!
+//! Utilities for graphQL backend v1 at <https://graphql.minaexplorer.com/>
+//!
+
+use crate::processor::*;
+use mina_serialization_types::json::*;
+use serde::{Deserialize, Serialize};
+use wasm_bindgen::prelude::*;
+
+#[wasm_bindgen(module = "/js/graphql_v1_utils.js")]
+extern "C" {
+    #[wasm_bindgen(catch)]
+    pub async fn fetch_block_json_str(height: usize, state_hash: &str) -> Result<JsValue, JsValue>;
+
+    #[wasm_bindgen(catch)]
+    pub async fn query_latest_blocks_json_str(limit: usize) -> Result<JsValue, JsValue>;
+}
+
+/// Fetch [ExternalTransitionJson] with its height and state hash
+pub async fn fetch_block(
+    height: usize,
+    state_hash: &str,
+) -> anyhow::Result<ExternalTransitionJson> {
+    let json_str = fetch_block_json_str(height, state_hash)
+        .await
+        .map_err(|err| anyhow::Error::msg(format!("{:?}", err)))?
+        .as_string()
+        .unwrap_or_default();
+    Ok(serde_json::from_str(&json_str)?)
+}
+
+/// Query infomation of the latest blocks
+pub async fn query_latest_blocks(limit: usize) -> anyhow::Result<Vec<BlockBasicInfo>> {
+    let json_str = query_latest_blocks_json_str(limit)
+        .await
+        .map_err(|err| anyhow::Error::msg(format!("{:?}", err)))?
+        .as_string()
+        .unwrap_or_default();
+    Ok(serde_json::from_str(&json_str)?)
+}
+
+/// Basic block information as query result
+#[derive(Debug, Clone, Serialize, Deserialize)]
+pub struct BlockBasicInfo {
+    /// Block height
+    #[serde(rename = "blockHeight")]
+    pub block_height: usize,
+
+    /// Block state hash
+    #[serde(rename = "stateHash")]
+    pub state_hash: String,
+}
+
+/// TODO: Doc
+#[derive(Debug, Clone, Default)]
+pub struct NonConsensusGraphQLV1Backend {
+    block_responder: Option<mpsc::Sender<ExternalTransitionJson>>,
+}
+
+impl NonConsensusGraphQLV1Backend {
+    /// TODO: Doc
+    pub fn new() -> Self {
+        Default::default()
+    }
+
+    /// TODO: Doc
+    pub async fn poll_latest_once(&self) -> anyhow::Result<()> {
+        if let Some(block_responder) = &self.block_responder {
+            let blocks = query_latest_blocks(10).await?;
+            for b in blocks {
+                if let Ok(block) = fetch_block(b.block_height, b.state_hash.as_str()).await {
+                    _ = block_responder.send(block).await;
```

Comment:
add `?` at the end?

Reply:
error should not break the loop here, use `log::error!` here instead